### ISSUE TITLE
Env loading is brittle

### ISSUE THREAD

**Issue by gabriel** [keybase/client#734]
You'll notice the NewHomeFinder taking functions because config is null initially.

https://github.com/keybase/client/blob/devel-mode/go/libkb/env.go#L166-L176

Also the Env type loads config from functions.

Instead of loading a NullConfiguration and then having it filled, we should load an environment with a set config in the constructor and have it be immutable. Then you don't need all these lazy loaded functions.

This also would prevent config from being changed while the program is running.

Basically, the environment and config should load on startup and remain immutable.
